src/qibolab/instruments/qm/config/config.py

```diff
@@ -0,0 +1,125 @@
+from dataclasses import dataclass, field
+from typing import Union
+
+from qibolab.components.configs import IqConfig, OscillatorConfig
+from qibolab.pulses import Pulse
+
+from ..components import OpxOutputConfig, QmAcquisitionConfig, QmChannel
+from .devices import *
+from .elements import *
+from .pulses import *
+
+__all__ = ["QmConfig"]
+
+
+@dataclass
```

Comment:
Since this is fully internal, let me postpone this and the other pydantic related comment to a later release: https://github.com/qiboteam/qibolab/issues/967.